src/main/javadoc/overview.html

```diff
@@ -0,0 +1,362 @@
+        <strong>com.amazonaws.kinesisvideo.java.mediasource.file.ImageFileMediaSource</strong></li>
+</ul>
+
+<div>
+    <h3>Prerequisites</h3>
+    <ul>
+        <li><strong>Java:</strong> Version 11, 17, or 21</li>
+        <li><strong>Maven:</strong> For building the project</li>
+        <li><strong>AWS Credentials:</strong> Access key and secret key</li>
+        <li><strong>Native Library:</strong> Platform-specific JNI library (provided)</li>
+    </ul>
+
+    <h3>Pre-built Libraries Available</h3>
+    <p>We provide pre-built native libraries for:</p>
+    <ul>
+        <li>macOS (Intel &amp; Apple Silicon)</li>
+        <li>Ubuntu Linux (x64)</li>
+        <li>Windows 10+</li>
+        <li>Raspberry Pi</li>
+    </ul>
+    <p>Set the <code>java.library.path</code> JVM property to the <i>directory</i> containing the JNI library.</p>
+
+    <pre style="background-color: #f4f4f4; padding: 15px; border-left: 4px solid #2196f3; margin: 10px 0; font-family: 'Courier New', monospace;">// Pre-built libraries are located in:
+src/main/resources/lib/
+├── mac/
+├── ubuntu/
+├── windows/
+└── raspberry-pi/</pre>
+
+    <div style="background-color: #ffebee; padding: 15px; border-left: 4px solid #f44336; margin: 15px 0;">
+        <strong>Important:</strong> If your platform isn't listed, you'll need to build the JNI library yourself.
+        See the <a
+            href="https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-java/wiki/Building-and-Using-JNI">building
+        guide</a>.
+    </div>
+</div>
+
+<h3>Running the Demo</h3>
+<pre style="background-color: #f4f4f4; padding: 15px; border-left: 4px solid #2196f3; margin: 10px 0; font-family: 'Courier New', monospace;"># Build the project
+mvn clean compile assembly:single
+
+# Run the demo (replace with your values)
+java -classpath target/*jar-with-dependencies.jar \
+  -Daws.accessKeyId=YOUR_ACCESS_KEY \
+  -Daws.secretKey=YOUR_SECRET_KEY \
+  -Dkvs-stream=YOUR_STREAM_NAME \
+  -Djava.library.path=src/main/resources/lib/mac \
+  -Dlog4j.configurationFile=log4j2.xml \
+  com.amazonaws.kinesisvideo.demoapp.DemoAppMain</pre>
+
+<h2>Learn More</h2>
+<p>Click on any section below to dive deeper into the topics that interest you:</p>
+
+<details>
+    <summary>Result Handling &amp; State Transitions (How PIC Decides Next Steps)</summary>
+    <div>
+        <h3>The Critical Pattern: Results Drive State Transitions</h3>
+        <p>
+            <strong>Summary:</strong> Every KVS API call result (HTTP status codes, success/failure)
+            determines what PIC does next.
+        </p>
+
+        <div style="background-color: #e8f5e8; padding: 15px; border-left: 4px solid #4caf50; margin: 15px 0;">
+            <strong>How it Works:</strong>
+            <ol>
+                <li>PIC calls JNI, which calls Java for API calls (e.g., describeStream)</li>
+                <li>Java makes HTTP call to AWS and gets result</li>
```

Comment:
(nit) Adding formatting like `result` or _result_ or changing to "a result" might be more clear